lamps-rfc7030-csrattrs.mkd

```diff
@@ -316,17 +317,22 @@ Otherwise, the `subjectPublicKey` MUST be absent.
 is defined where the '`extnValue`' field has been made `OPTIONAL`.
 This is only needed to enable specifying partial extensions with values to be filled in
 by the client; otherwise the `id-ExtensionReq` OID and the respective value of type
-`ExtensionReq` MUST be used.<br>
+`ExtensionReq` MUST be used.
 
 For each extension of type `Extension` or `ExtensionTemplate` provided by the server,
 the client is expected to include an extension of the type given by the `extnID`.
-If the '`critical`' field is present, the client is expected to use the given value.
+If the 'critical' field is present, the client SHOULD use it in the extension as well.
 If the '`extnValue`' is present (which is always the case when type `Extension` is used),
-the client is required to use the given extension value.
-Otherwise it is expected to fill in the extension value.<br>
-If the '`subjectAltName`' extension contains the '`directoryName`' choice containing the `NULL-DN`
+the client MUST use the given extension value in its CSR.
+When the type `ExtensionTemplate` is used, the '`extnValue`' can be absent, and then the client SHOULD provide an extension value in an extension with the given `extnID`.
+For instance, if the server includes an `ExtensionTemplate` with the `extnID` '`subjectAltName`' but without an `extnValue`, the client SHOULD add a SAN extension (with same suitable value).
+
+In case the server includes an `ExtensionTemplate` with the `extnID` '`subjectAltName`'
+and a partially filled in `extnValue`, such as a '`directoryName`' choice containing the `NULL-DN`
 (i.e., an empty sequence of RDNs) or the '`iPAddress`' choice with an empty `OCTET STRING`,
-this means that the client is expected to fill in the respective `GeneralName` value.
+this means that the client SHOULD fill in the respective `GeneralName` value.
+
+When the a different type is used, then the extnValue MAY be empty, and in that case the client is expected to provide a value in an extension of that type.
```

Comment:
In this line contains old invalid text got copied in, which should simply be removed.